packages/office-ui-fabric-react/src/components/TextField/TextField.tsx

```diff
@@ -417,6 +542,22 @@ export class TextField extends BaseComponent<ITextFieldProps, ITextFieldState> i
     onBeforeChange(value);
   }
 
+  @autobind
+  private _onKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
+    const key = event.keyCode || event.charCode;
+
+    // Check for backspace
+    if (key == 8) {
+      if (this.props.mask && this.state.maskCursorPosition && this.state.value) {
+        if ((event.target as HTMLInputElement).selectionStart >= this.state.maskCursorPosition) {
```

Comment:
If you select multiple characters and backspace them, the inputChanged event will detect that you deleted a bunch of characters. Like if you ctrl+a and backspace, it will clear any input the user has made and display the empty mask.